### PR TITLE
config,docs: Impl default and add more docs

### PR DESCRIPTION
**Pull request by Hoverbear**
Also some `rustfmt` related changes.

Most defaults were taken from TiKV.

### CODE REVIEW

File: src/raft.rs [hunk 1]
```diff
@@ -147,7 +146,39 @@ pub struct Config {
...
+            max_election_tick: 0,
+            read_only_option: ReadOnlyOption::Safe,
+            skip_bcast_commit: false,
+            tag: "".into(),
+        }
+    }
+}
+
 impl Config {
+    pub fn new(id: impl Into<u64>) -> Self {
```
[the remark this answers is not in the document]
Comment (Hoverbear) [reply src/raft.rs]: Also it breaks CI :)

**Comment by breeswish** [comment]
@overvenus PTAL

File: src/raft.rs [hunk 2]
```diff
@@ -147,7 +146,37 @@ pub struct Config {
     pub tag: String,
 }
 
+impl Default for Config {
+    fn default() -> Self {
+        const HEARTBEAT_TICK: usize = 150;
```
Comment (overvenus) [inline src/raft.rs]: Looks like we need to tick 150 times to send a heartbeat, maybe too long.

File: src/lib.rs [hunk 3]
```diff
@@ -39,41 +39,53 @@ use raft::{
...
+use std::{sync::mpsc::{channel, RecvTimeoutError}, time::{Instant, Duration}};
+
+// We're using a channel, but this could be any stream of events.
+let (tx, rx) = channel();
+let timeout = Duration::from_millis(100);
+
+// Send the `tx` somewhere else...
+
+let ticks = 5; // Only tick 5 times.
+let mut current_timer = timeout;
```
Comment (siddontang) [inline src/lib.rs]: is current_timer a timer?

Comment (Hoverbear) [reply src/lib.rs]: No, this is misnamed.

File: src/raft.rs (hunk 2 above)
Comment (Hoverbear) [reply src/raft.rs]: TiKV uses 2, I'll update it to that. https://github.com/pingcap/tikv/blob/2b3f52d35adebd8d95fa052dd58b33338a44aae0/src/raftstore/store/config.rs#L138

File: src/raft.rs [hunk 4]
```diff
@@ -147,7 +146,37 @@ pub struct Config {
...
+            skip_bcast_commit: false,
+            tag: "".into(),
+        }
+    }
+}
+
 impl Config {
+    pub fn new(id: u64) -> Self {
+        Self {
+            id,
```
Comment (siddontang) [inline src/raft.rs]: why not setting tag to `format!("{}", id)`?

Comment (Hoverbear) [reply src/raft.rs]: Oh, great idea! :+1:

**Comment by Hoverbear** [comment]
Related to #70

**Pull request merged**